test/test_gradient.jl

```diff
@@ -169,6 +169,11 @@ T = GradientTracer{P}
             ]
         end
 
+        @testset "Multiplication by zero" begin
+            f(x) = Matrix(I(length(x))) * x
+            @test J(f, ones(10)) == I(10)
+        end
+
```

Comment:
Can we replace this by a simpler test? 
This is bad code nobody should ever write.

Reply:
Same goes for the copy in `test_hessian`.